mars_form_app/src/components/Form.js

```diff
@@ -0,0 +1,321 @@
+import React, {Component} from 'react';
+// import axios from 'axios';
+import countries from "./countries"
+
+class Form extends Component{
+    state = {
+        name: "",
+        birthday: "",
+        countries: countries,
+        selectedCountry: "United States",
+        diet: "",
+        explorer: "",
+        formCompleted: false,
+        formSubmitted: false,
+        breathe:"no",
+        married:false,
+        situation:"",
+        claustrophobic:false,
+        familyHistory:{},
+        living:{},
+        education:{}
+    }
+
+    // populateCountry() {
+    //     // e.preventDefault();
+    //     axios.get("http://vocab.nic.in/rest.php/country/json").then(res => {
+    //         let {countries} = res.data;
+    //         // debugger
+    //         // this.setState({countries});
+    //     })
+    // }
+    handleChange = (e) => {
+        this.setState({
+            [e.target.name]: e.target.value
+        })
+    }
+    handleSubmit = (e) => {
+        e.preventDefault();
+        this.setState({
+            formCompleted: !this.state.formCompleted
+        })
```

Comment:
when referring to previous state setState should be passed a the callback function.

Reply:
You probably just want to set forCompleted to true. Otherwise it could keep flipping back and forth with multiple submits.